packages/flutter/lib/src/material/slider_theme.dart

```diff
@@ -323,6 +338,24 @@ class SliderThemeData extends Diagnosticable {
   /// The color given to the [valueIndicatorShape] to draw itself with.
   final Color valueIndicatorColor;
 
+  /// The shape and behavior that will be used to draw the [Slider]'s track.
+  ///
+  /// This can be customized by implementing a subclass of
```

Comment:
I was following the pattern of the other existing components here. If I remove "This can be customized by implementing a subclass of Foo" from the 3 new components, should I also take it away from the existing 2?

Reply:
> It might be worth pointing out how he trackShape's preferred size contribute's to the slider's overall preferred size.

It doesn't actually do that in this change.

The default track size paints relative to the size of the parent box, but it subtracts out the size largest overlapping piece: the overlay. The intrinsic sizes are dependent on the overlay, thumb, and value indicators. The track shape's preferredRect is used inside the Slider class for:

1. Tap gesture points to thumb position range, where 0.0 is track.left and 1.0 is track.right.
2. The thumb center and discrete tick centers for their shape painters.

I added these as doc comments, but will go in more detail once I get to the actual SliderTrackShape doc comments.

> It would be worth pointing out that the default value of trackShape is RectangularSliderTrackShape.

Done